services/src/messaging/src/main/java/org/openkilda/messaging/model/SwitchId.java

```diff
@@ -0,0 +1,157 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.model;
+
+import org.openkilda.messaging.error.SwitchIdFormatException;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents a switch id entity.
+ */
+public class SwitchId implements Serializable, Comparable<SwitchId> {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Switch switchId.
+     */
+    private final long switchId;
+
+    /**
+     * Instance constructor.
+     *
+     * @param switchId switch id type string.
+     */
+    public SwitchId(String switchId) {
+        long value;
+        try {
+            value = Long.parseUnsignedLong(switchId.replaceAll("[-:]", ""), 16);
+        } catch (NumberFormatException e) {
+            throw new SwitchIdFormatException(String.format("Can not parse input string: \"%s\"", switchId));
+        }
+
+        this.switchId = value;
+    }
+
+    /**
+     * Instance constructor.
+     *
+     * @param switchId switch id type long.
+     */
+    public SwitchId(long switchId) {
+        this.switchId = switchId;
+    }
+
+    /**
+     * Return the switch id in long.
+     *
+     * @return the switch id in long.
+     */
+    public long toLong() {
+        return switchId;
+    }
+
+    /**
+     * Return the MAC address of switch.
+     *
+     * @return the MAC address of switch.
+     */
+    public String toMacAddress() {
+        return colonSeparatedBytes(toHexArray(), 4);
+    }
+
+    /**
+     * {@inheritDoc }
+     */
+    @Override
+    @JsonValue
+    public String toString() {
+        return colonSeparatedBytes(toHexArray(), 0);
+    }
+
+    /**
+     * Return the switch id in otsd format.
+     *
+     * @return the switch id in otsd format.
+     */
+    public String toOtsdFormat() {
+        return "SW" + new String(toHexArray()).toUpperCase();
+    }
+
+    private String colonSeparatedBytes(char[] hex, int offset) {
```

Comment:
This method must be covered with unit tests.

Reply:
Tests have been added.